### PR TITLE
Fix compilation + mldb_sample_plugin.mk

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: plugin.cc [hunk 1]
```diff
@@ -4,9 +4,10 @@
    Copyright (c) 2015 Datacratic Inc.  All rights reserved.
  */
 
-#include "mldb/soa/types/value_description.h"
-#include "mldb/server/function.h"
-#include "mldb/server/plugin.h"
+#include "mldb/types/value_description.h"
+#include "mldb/core/function.h"
+#include "mldb/core/plugin.h"
+#include "mldb/rest/poly_entity.h"
```
Comment (FinchPowers) [inline plugin.cc:10]: It is bizarre to require part of the rest package to have struct Package.

File: load_plugin_demo.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+hello_world_url = '/v1/types/functions/helloWorld'
+
+res = mldb.perform('GET', hello_world_url)
+assert res['statusCode'] == 404, str(res)
+
+perform('PUT', '/v1/plugins/sample', [], {
+    'type' : 'sharedLibrary',
+    'params' : {
+        'address' : '/home/fmlheureux/workspace/mldb/build/x86_64/lib/',
```
Comment (FinchPowers) [inline load_plugin_demo.py]: I will fix that ;)

File: plugin.cc [hunk 3]
```diff
@@ -103,13 +104,14 @@ MLDB::Plugin * mldbPluginEnter(MldbServer * server)
...
-static RegisterFunctionType<HelloWorldFunction, int>
-regHelloWorldFunction("hello.world",
-                    "Sample function that always returns hello = \"world\"",
-                    "No documentation available");
-
+Package package("mldb_sample_plugin");
 
+static RegisterFunctionType<HelloWorldFunction, int>
+regHelloWorldFunction(package,
+                      "helloWorld",
```
Comment (guyd) [inline plugin.cc:114]: very minor but I prefer hello.world or sample.plugin since it follows our type naming convention (e.g. classifier.train)

Comment (FinchPowers) [inline plugin.cc:114]: Then how does it work with the sql? Is hello.world a valid function call?

Comment (guyd) [inline plugin.cc:114]: isn't the function type?  So you will not invoke hello.world but a function of that type.  If that is not the case then forget my comment.

Comment (FinchPowers) [inline plugin.cc:114]: I don't understant. With the current form I expect something like

```
SELECT helloWorld(1)
```

With the form hello.world, would the call be

```
SELECT hello.world(1)
```

?

Comment (guyd) [inline plugin.cc:114]: RegisterFunctionType registers a function type not a function name (here helloWord).  So you create a function of type sample.plugin:

```
PUT /v1/functions/helloWorld {id : sample.plugin}
```

and you call `SELECT helloWorld(args)`

**Comment by guyd** [comment]
+1

**Pull request merged**